runlike/inspector.py

```diff
@@ -201,7 +214,11 @@ def format_cli(self):
             parameters += self.options
         parameters.append(image)
 
-        cmd_parts = self.get_fact("Config.Cmd")
+        entrypoint_parts = self.get_fact("Config.Entrypoint") or []
+        if len(entrypoint_parts) > 1:
+            entrypoint_parts = entrypoint_parts[1:]
+        cmd_parts = entrypoint_parts + (self.get_fact("Config.Cmd") or [])
```

Comment:
Is this really how the entrypoint feature works? Does it ever have more than one part?
e.g. the [docs](https://docs.docker.com/engine/reference/run/) say:
```
Note: Passing --entrypoint will clear out any default command set on the image (i.e. any CMD instruction in the Dockerfile used to build it).
```

I think we need to add a bit more testing to cover the full range of interaction between entrypoint and cmd.
So, 2x2 possibilities: ec, eC, Ec, EC (where E: entrypoint was specified, e: no entrypoint, C: cmd was specified, c: no cmd). Your test already covers EC. It almost covers ec (if you test for the absence of `--entrypoint` in the output).

Reply:
Hm, you may be right. This is what I was going by:
https://aws.amazon.com/blogs/opensource/demystifying-entrypoint-cmd-docker/

About halfway down the page:

> Suppose, as before, we have the following Dockerfile and create an image from it called myservice:

```

ENTRYPOINT ["/bin/chamber", "exec", "production", "--"]
CMD ["/bin/service", "-d"]
```

> Now, let’s change the ENTRYPOINT by running the following:
> 
> docker run --entrypoint /bin/logwrap myservice
> Per our general rule, the following argument list will be constructed:
> 
`
["/bin/logwrap", "/bin/service", "-d"]
`
But I just gave that a try with this Dockerfile:

```
FROM ubuntu:latest
ENTRYPOINT ["/bin/bash"]
CMD ["-c", "ls"]
```
running it like
```
docker build . -t test
docker run --rm test
```
gives me the `ls` of the root, but running it like

```
docker run --rm --entrypoint /bin/bash test
```
gives me no output at all.

I'll have to investigate this further before making a fix

Reply:
> So, 2x2 possibilities: ec, eC, Ec, EC (where E: entrypoint was specified, e: no entrypoint, C: cmd was specified, c: no cmd). Your test already covers EC. It almost covers ec (if you test for the absence of `--entrypoint` in the output).

The thing is, you specified an `ENTRYPOINT` in the fixture `Dockerfile`, so there's never going to be a situation with the absence of `--entrypoint`.

Reply:
Wow, this went unnoticed for such a long time. Sorry about that. 

Yea, you're right about the Dockerfile's entrypoint. Feel free to change it or add an other fixture.